src/GraphQL/StringExtensions.cs

```diff
@@ -119,20 +119,26 @@ public static Inputs ToInputs(this JObject obj)
         /// <summary>
         /// Returns a camel case version of the string.
         /// </summary>
+        /// <param name="s">The source string.</param>
+        /// <returns>System.String.</returns>
         public static string ToCamelCase(this string s)
         {
             if (string.IsNullOrWhiteSpace(s))
             {
                 return string.Empty;
             }
 
-            return $"{char.ToLowerInvariant(s[0])}{s.Substring(1)}";
+            var newFirstLetter = char.ToLowerInvariant(s[0]);
+            if (newFirstLetter == s[0])
+                return s;
+
+            return newFirstLetter + s.Substring(1);
```

Comment:
I would like to pull in this PR but I'm concerned that adding these additional checks will slow down these methods.  Was there a reason for adding these additional checks?  Did you do perf tests and see that this was faster?

https://www.dotnetperls.com/uppercase-first-letter

The above suggests to switch to the following for higher performance:

```csharp
char[] a = s.ToCharArray();
a[0] = char.ToUpper(a[0]);
return new string(a);
```

Reply:
Oops, I accidentally clicked cancel review.

Reply:
OK, for the interest I have conducted the tests.
Demo:
```c#
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;

namespace ConsoleApp6
{
    [MemoryDiagnoser]
    public class Program
    {
        [Params("someName", "SomeName")]
        public string s;

        [Benchmark(OperationsPerInvoke = 16)]
        public string ToCamelCaseOld()
        {
            if (string.IsNullOrWhiteSpace(s)) return string.Empty;

            return $"{char.ToLowerInvariant(s[0])}{s.Substring(1)}";
        }

        [Benchmark(OperationsPerInvoke = 16)]
        public string ToCamelCaseNew()
        {
            if (string.IsNullOrWhiteSpace(s)) return string.Empty;

            var newFirstLetter = char.ToLowerInvariant(s[0]);
            if (newFirstLetter == s[0])
                return s;

            return newFirstLetter + s.Substring(1);
        }

        [Benchmark(OperationsPerInvoke = 16)]
        public string ToCamelCaseWithArray()
        {
            if (string.IsNullOrWhiteSpace(s)) return string.Empty;

            char[] a = s.ToCharArray();
            a[0] = char.ToLowerInvariant(a[0]);
            return new string(a);
        }

        static void Main(string[] args)
        {
            var summary = BenchmarkRunner.Run<Program>();
        }
    }
}
```

Results:

``` ini

BenchmarkDotNet=v0.11.4, OS=Windows 10.0.17134.648 (1803/April2018Update/Redstone4)
Intel Core i5-6200U CPU 2.30GHz (Skylake), 1 CPU, 4 logical and 2 physical cores
Frequency=2343750 Hz, Resolution=426.6667 ns, Timer=TSC
.NET Core SDK=2.2.104
  [Host]     : .NET Core 2.1.8 (CoreCLR 4.6.27317.03, CoreFX 4.6.27317.03), 32bit RyuJIT
  DefaultJob : .NET Core 2.1.8 (CoreCLR 4.6.27317.03, CoreFX 4.6.27317.03), 32bit RyuJIT


```
|               Method |        s |      Mean |     Error |    StdDev |    Median | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
|--------------------- |--------- |----------:|----------:|----------:|----------:|------------:|------------:|------------:|--------------------:|
|       **ToCamelCaseOld** | **SomeName** | **12.975 ns** | **0.5710 ns** | **1.6474 ns** | **12.642 ns** |      **0.0035** |           **-** |           **-** |                 **6 B** |
|       ToCamelCaseNew | SomeName |  5.013 ns | 0.2041 ns | 0.5791 ns |  4.943 ns |      0.0030 |           - |           - |                 5 B |
| ToCamelCaseWithArray | SomeName |  3.533 ns | 0.1134 ns | 0.3254 ns |  3.504 ns |      0.0024 |           - |           - |                 4 B |
|       **ToCamelCaseOld** | **someName** | **12.883 ns** | **0.3666 ns** | **1.0517 ns** | **12.943 ns** |      **0.0035** |           **-** |           **-** |                 **6 B** |
|       ToCamelCaseNew | someName |  1.038 ns | 0.0311 ns | 0.0906 ns |  1.025 ns |           - |           - |           - |                   - |
| ToCamelCaseWithArray | someName |  3.357 ns | 0.1235 ns | 0.3562 ns |  3.153 ns |      0.0024 |           - |           - |                 4 B |

So, in any case, the proposed method impl is much faster than the current one. And if the word begins with a lowercase letter, then there is no allocation at all. Additional character checking is nothing compared to the time spent on allocating objects :).